src/reconcile/rec_visibility.c

```diff
@@ -331,8 +333,28 @@ __rec_find_and_save_delete_hs_upd(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT
             if (!seen_committed)
                 break;
             WT_RET(__rec_delete_hs_upd_save(session, r, ins, rip, delete_upd, NULL));
+            visible_all_upd = NULL;
             break;
         }
+
+        /*
+         * Track the first self-contained value that is globally visible. If we free the updates
+         * before the tombstone, due to the tombstone being globally visible concurrently with the
```

Comment:
What is the tombstone here? I don't understand this comment.

Reply:
The comments are updated, this to handle the prepare rollback scenario.